### PR TITLE
Chore/#33 remove g lua ifdefs

### PR DESCRIPTION
**Pull request by MJLHThomassen**
I'm gonna go ahead and PR this. The Signal 11 bug when g_debugLua is 1 will be fixed as part of #30

Sorry for the double PR, github PR is a bitch sometimes... (need to select the correct repo first, then the correct base, then the correct compare, meh, its a drag)

### CODE REVIEW

File: code/cgame/cg_lua.c [hunk 1]
```diff
@@ -356,6 +354,4 @@ lvm_t *CG_LuaGetVM(lua_State * L)
         if (lVM[i] && lVM[i]->L == L)
             return lVM[i];
     return NULL;
-}
-
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/cgame/cg_lua.c]: newline

File: code/cgame/lua_cent.c [hunk 2]
```diff
@@ -70,5 +70,4 @@ cent_t *Lua_GetCent(lua_State * L, int argNum)
     ud = luaL_checkudata(L, argNum, "cgame.centity");
     luaL_argcheck(L, ud != NULL, argNum, "\'centity\' expected");
     return (cent_t *)ud;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/cgame/lua_cent.c]: newline

File: code/cgame/lua_cfx.c [hunk 3]
```diff
@@ -15,5 +13,4 @@ void Lua_CFX_LoadMapFxFile(void) {
     if (!file) return;
 
     Lua_CFX_ParseMapFxFile(&file);
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/cgame/lua_cfx.c]: newline

File: code/cgame/lua_refent.c [hunk 4]
```diff
@@ -467,7 +463,4 @@ rent_t *Lua_GetRent(lua_State * L, int argNum)
     ud = luaL_checkudata(L, argNum, "cgame.refentity");
     luaL_argcheck(L, ud != NULL, argNum, "\'refentity\' expected");
     return (rent_t *)ud;
-}
-#endif
-
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/cgame/lua_refent.c]: newline

File: code/game/Makefile [hunk 5]
```diff
@@ -164,7 +164,7 @@ else
 DO_SOCC = $(CC) $(SOCFLAGS) $(WL) $(DEFINES) -o $@ -c $<
 endif
 # do cc for lua
-ifeq ($(DEBUF), 1)
+ifeq ($(DEBUG), 1)
```
Comment (solarisstar) [inline code/game/Makefile:167]: LOL seriously?

File: code/game/g_lua.c [hunk 6]
```diff
@@ -1118,6 +1116,4 @@ void G_LuaCollectGarbage(void) {
             lua_gc(vm->L, LUA_GCCOLLECT, 0);
         }
     }
-}
-
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/g_lua.c]: newline

File: code/game/g_main.c [hunk 7]
```diff
@@ -2760,9 +2742,5 @@ void G_RunFrame(int levelTime) {
         }
     }
 
-    //RPG-X: Marcin: To clear pressedUse. - 30/12/2008
-
-#ifdef G_LUA
     LuaHook_G_RunFrame(levelTime);
-#endif
 }
```
Comment (solarisstar) [inline code/game/g_main.c]: newline

File: code/game/lua_cvar.c [hunk 8]
```diff
@@ -115,5 +113,4 @@ int Luaopen_Cvar(lua_State *L) {
     luaL_register(L, "cvar", lib_cvar);
 
     return 1;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_cvar.c]: newline

File: code/game/lua_game.c [hunk 9]
```diff
@@ -478,5 +476,4 @@ int Luaopen_Game(lua_State *L) {
     lua_pushliteral(L, GAMEVERSION);
 
     return 1;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_game.c]: newline

File: code/game/lua_mover.c [hunk 10]
```diff
@@ -579,5 +582,4 @@ static const luaL_Reg lib_mover[] = {
 int Luaopen_Mover(lua_State *L) {
     luaL_register(L, "mover", lib_mover);
     return 1;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_mover.c]: newline

File: code/game/lua_sound.c [hunk 11]
```diff
@@ -44,5 +42,4 @@ static const luaL_Reg lib_sound[] = {
 int Luaopen_Sound(lua_State *L) {
     luaL_register(L, "sound", lib_sound);
     return 1;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_sound.c]: newline

File: code/game/lua_trace.c [hunk 12]
```diff
@@ -255,5 +253,4 @@ ltrace_t *Lua_GetTrace(lua_State * L, int argNum)
     ud = luaL_checkudata(L, argNum, "game.trace");
     luaL_argcheck(L, ud != NULL, argNum, "\'trace\' expected");
     return (ltrace_t *)ud;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_trace.c]: newline

File: code/game/lua_vector.c [hunk 13]
```diff
@@ -446,6 +444,4 @@ int Lua_IsVector(lua_State * L, int idx)
         }
     }
     return 0;
-}
-
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_vector.c]: newline

File: code/game/lua_weapons.c [hunk 14]
```diff
@@ -118,5 +116,4 @@ int Luaopen_Weapons(lua_State *L) {
     luaL_register(L, "weapons", lib_weapons);
 
     return 1;
-}
-#endif
\ No newline at end of file
+}
```
Comment (solarisstar) [inline code/game/lua_weapons.c]: newline

**Comment by solarisstar** [comment]
Let's discuss what we're going to do about brackets going forward.

**Comment by MJLHThomassen** [comment]
What's with the newlines? Do they need to be there?

File: code/game/Makefile (hunk 5 above)
Comment (MJLHThomassen) [inline code/game/Makefile:167]: _sigh_ yes

**Comment by solarisstar** [comment]
It has to do with how POSIX defines a file. Unix Tools expect every line in a file to end in a newline character, including the last line. It's not super essential, but it makes using command line tools a little less complicated.

**Comment by MJLHThomassen** [comment]
Right ill get on that then and remember it for the future.

**Comment by solarisstar** [comment]
Yeah, I know it seems trivial but it's all-around nicer.

**Pull request merged**